manager/integration/tests/test_ha.py

```diff
@@ -2141,6 +2141,46 @@ def test_auto_remount_with_subpath(client, core_api, storage_class, sts_name, st
     assert expect_md5sum == md5sum
 
 
+@pytest.mark.skip(reason="WAIT_FOR_UPSTREAM_FIX") # NOQA
+@pytest.mark.skipif( "k3s" not in k8s_api_client.status.node_info.kubelet_version, reason='This test is only compaitable with k3s')  # NOQA
+def test_restart_kubelete_when_pod_with_subpath():
+    """
+    Related issue 3080:
+    https://github.com/longhorn/longhorn/issues/3080
+
+    Steps:
+    0. Check if the cluster runs with k3s
+    1. Create deployment with subpath with pods on the same targeting-node
+    2. Wait for pods become Running
+    3. Restart k3s kubelet-agent service
+    4. Wait for new pods become Running
+    5. Check kubelet log from the targeting-node
+    """
+
+    # Get first worker node
+    node = k8s_api_client.list_node().items[0]
+
+    # Add label to the node
+    node.metadata.labels['targeting-node'] = 'True'
+
+    # Create deployment with subpath
+    deployment = make_deployment_with_pvc("deployment-with-subpath", "pvc1", replicas=1)
+    deployment.spec.template.spec.containers[0].volume_mounts[0].sub_path = "sub"
+
+    # Wait for deployment become ready
+    wait_for_deployment_ready(k8s_api_client, deployment)
+    pod_name = deployment.spec.template.metadata.name
+
+    # TODO: Specify node to run the command
+
+    common.restart_k3s_agent
+    common.wait_for_pod_phase(core_api, pod_name, pod_phase="Running")
+
+    # Check kubelet log from the targeting-node
+    expect_restult = not common.check_k3s_agent_log("Path does not exist", 60*5)
+    assert expect_restult == True
+
```

Comment:
Assert the data of the volume.

Reply:
Added get_container_data_md5sum function to validate the volume data in the latest commit.